integrated_channels/canvas/client.py

```diff
@@ -505,12 +481,20 @@ def _handle_get_user_canvas_course(self, canvas_user_id, learner_data_course_id)
 
         return canvas_course_id
 
-    def _get_oauth_access_token(self, client_id, client_secret):
-        """Uses the client id, secret and refresh token to request the user's auth token from Canvas.
+    def _create_session(self):
+        """
+        Instantiate a new session object for use in connecting with Canvas. Each enterprise customer
+        connecting to Canvas should have a single client session.
+        Will only create a new session if token expiry has been reached
+        """
+        self.session, self.expires_at = refresh_session_if_expired(
+            self._get_oauth_access_token,
```

Comment:
Oh I see, you're able to make this a util for both Canvas and blackboard by passing in their individual token retrieval methods 👍 . I'm very in favor of this!

Reply:
yeah I had forgotten we had this method then reviewing side by side just saw they were identical as long as I can pasa f unctoin to get oauth_access_token so works out!